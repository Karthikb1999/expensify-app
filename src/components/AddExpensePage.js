import React from "react"
import ExpenseForm from "../components/ExpenseForm"
import { startAddExpense } from "../actions/expenses"
import { connect } from "react-redux"


export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense)
        this.props.history.push("/")
    }
    render() {
        return (
            <div>
                <h2>Add Expense</h2>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}


const matchDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => { dispatch(startAddExpense(expense)) }
})

export default connect(undefined, matchDispatchToProps)(AddExpensePage)