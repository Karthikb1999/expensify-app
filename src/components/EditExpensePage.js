import React from "react"
import ExpenseForm from "../components/ExpenseForm"
import { connect } from "react-redux"
import { startEditExpense, startRemoveExpense } from "../actions/expenses"


export class EditExpensePage extends React.Component {
    onSubmit = (changes) => {
        this.props.startEditExpense(this.props.expense.id, changes)
        this.props.history.push("/")
    }
    onClick = () => {
        this.props.startRemoveExpense(this.props.expense.id)
        this.props.history.push("/")
    }

    render() {
        return (
            <div>
                <ExpenseForm expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onClick}>Remove Expense</button>
            </div>
        )
    }

}


const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchtoProps = (dispatch) => ({
    startEditExpense: (id, changes) => {
        dispatch(startEditExpense(id, changes))
    },
    startRemoveExpense: (id) => {
        dispatch(startRemoveExpense(id))
    }
})


export default connect(mapStateToProps, mapDispatchtoProps)(EditExpensePage)