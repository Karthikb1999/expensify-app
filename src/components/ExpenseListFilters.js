import React from "react";
import { connect } from "react-redux"
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from "../actions/filters"
import { DateRangePicker } from "react-dates"


export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocus: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocus) => {
        this.setState(() => ({
            calendarFocus
        }))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        if (e.target.value === "amount") {
            this.props.sortByAmount()
        } else {
            this.props.sortByDate()
        }
    }
    render() {
        return (
            <div className="container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type="text"
                            value={this.props.filters.text}
                            placeholder="Search expenses"
                            className="text-input"
                            onChange={this.onTextChange} />
                    </div>
                    <div className="input-group__item">
                        <select
                            value={this.props.filters.sortBy}
                            onChange={this.onSortChange}
                            className="select"
                        >
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocus}
                            onFocusChange={this.onFocusChange}
                            isOutsideRange={() => false}
                            showClearDates={true}
                            numberOfMonths={1}
                        />
                    </div>
                </div>
            </div>
        )
    }
}


const mapStatetoProps = (state) => ({
    filters: state.filters
})

const mapDispatchtoProps = (dispatch) => ({
    setStartDate: (startDate) => {
        dispatch(setStartDate(startDate))
    },
    setEndDate: (endDate) => {
        dispatch(setEndDate(endDate))
    },
    sortByAmount: () => {
        dispatch(sortByAmount())
    },
    sortByDate: () => {
        dispatch(sortByDate())
    },
    setTextFilter: (text) => {
        dispatch(setTextFilter(text))
    }

})



export default connect(mapStatetoProps, mapDispatchtoProps)(ExpenseListFilters)