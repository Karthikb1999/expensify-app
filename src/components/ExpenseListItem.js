import React from "react";
import { Link } from "react-router-dom"



const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>{description}</Link>
        <p>Amount: {amount} Created At: {createdAt}</p>
    </div>
)

export default ExpenseListItem