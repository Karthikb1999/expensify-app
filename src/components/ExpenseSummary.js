import React from "react"
import numeral from "numeral"

export const ExpenseSummary = (props) => {
    return (
        <p>Viewing {props.expensesCount} expense{props.expensesCount !== 1 ? 's' : ''} totalling {numeral(props.expensesTotal / 100).format('$0, 0.00')}</p>
    )
}

export default ExpenseSummary