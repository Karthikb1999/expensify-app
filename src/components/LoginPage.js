import React from "react"
import { startLogin } from "../actions/auth"
import { connect } from "react-redux"


export const LoginPage = ({ startLogin }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__box__title">Expensify</h1>
                <p>Get your expenses under control</p>
                <button className="button" onClick={startLogin}>Login with Google</button>
            </div>
        </div>
    )
}

const mapDispatchtoProps = (dispatch) => ({
    startLogin: () => {
        dispatch(startLogin())
    }
})

export default connect(undefined, mapDispatchtoProps)(LoginPage)