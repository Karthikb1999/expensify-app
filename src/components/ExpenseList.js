import React from "react"
import { connect } from "react-redux"
import ExpenseListItem from "../components/ExpenseListItem"
import getVisibleExpenses from "../selectors/expenses"


export const ExpenseList = ({ expenses }) => (
    <div className="container">
        <div className="list-header">
            <div className="show-to-desktop">Expense</div>
            <div className="show-to-desktop">Amount</div>
            <div className="show-to-mobile">Expenses</div>
        </div>
        <div className="list-body">
            {expenses.length === 0 &&
                <div className="list-item list-item--message">
                    <p>No expenses</p>
                </div>
            }
            {expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)}
        </div>
    </div>
)

const MapStateToProps = (state) => ({
    expenses: getVisibleExpenses(state.expenses, state.filters),
}
)

export default connect(MapStateToProps)(ExpenseList)