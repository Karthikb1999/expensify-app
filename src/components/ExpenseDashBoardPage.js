import React from "react"
import ExpenseList from "../components/ExpenseList"
import ExpenseListFilters from "../components/ExpenseListFilters";


const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashBoardPage