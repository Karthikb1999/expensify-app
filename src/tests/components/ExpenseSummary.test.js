import { ExpenseSummary } from "../../components/ExpenseSummary"
import { shallow } from "enzyme"
import React from "react"

test("should render summary with single summary correctly", () => {
    const expensesCount = 1
    const expensesTotal = 100
    const wrapper = shallow(<ExpenseSummary
        expensesCount={expensesCount}
        expensesTotal={expensesTotal}
    />)
    expect(wrapper).toMatchSnapshot()
})

test("should render summary with multi summary correctly", () => {
    const expensesCount = 5
    const expensesTotal = 1000
    const wrapper = shallow(<ExpenseSummary
        expensesCount={expensesCount}
        expensesTotal={expensesTotal}
    />)
    expect(wrapper).toMatchSnapshot()
})