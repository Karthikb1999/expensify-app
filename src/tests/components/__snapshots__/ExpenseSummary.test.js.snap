// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary with multi summary correctly 1`] = `
<p>
  Viewing 
  5
   expense
  s
   totalling 
  $10.00
</p>
`;

exports[`should render summary with single summary correctly 1`] = `
<p>
  Viewing 
  1
   expense
   totalling 
  $1.00
</p>
`;
